raiden/blockchain/events.py

```diff
@@ -251,20 +254,45 @@ class BlockchainEvents(object):
     def __init__(self):
         self.event_listeners = list()
 
-    def poll_all_event_listeners(self):
+    def poll_all_event_listeners(self, from_block=None):
         result = list()
+        reinstalled_filters = False
 
-        for event_listener in self.event_listeners:
-            decoded_events = poll_event_listener(
-                event_listener.filter,
-                event_listener.translator,
-            )
-            result.extend(decoded_events)
+        while True:
+            try:
+                for event_listener in self.event_listeners:
+                    decoded_events = poll_event_listener(
+                        event_listener.filter,
+                        event_listener.translator,
+                    )
+                    result.extend(decoded_events)
+                break
+            except EthNodeCommunicationError as e:
+                # If the eth client has restarted and we reconnected to it then
+                # filters will no longer exist there. In that case we will need
+                # to recreate all the filters.
+                if not reinstalled_filters and str(e) == 'filter not found':
+                    result = list()
+                    reinstalled_filters = True
+                    updated_event_listerners = list()
+
+                    for event_listener in self.event_listeners:
+                        new_listener = EventListener(
+                            event_listener.event_name,
+                            event_listener.filter_creation_function(from_block=from_block),
```

Comment:
Is the `from_block` inclusive?

I believe the value of `self._blocknumber` from `RaidenService` will the current block, which is being polled for events. 

Note: This actually depends on the [order of which the callbacks are installed](https://github.com/raiden-network/raiden/blob/3f8758c85fa84bbf0ae8ca68d2ae8990dde60a5e/raiden/raiden_service.py#L278) with the [alarm task](https://github.com/raiden-network/raiden/blob/3f8758c85fa84bbf0ae8ca68d2ae8990dde60a5e/raiden/tasks.py#L105), if the polling is executed first then `self._blocknumber` is the previous block.

Reply:
As of [this](https://ethereum.stackexchange.com/a/8219) it is inclusive, not sure about parity though.

Reply:
Btw, could you add a test for this under the [assumptions](https://github.com/raiden-network/raiden/blob/3f8758c85fa84bbf0ae8ca68d2ae8990dde60a5e/raiden/tests/integration/rpc/test_assumptions.py)?